clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/agent/CatsClusterCachingAgent.java

```diff
@@ -93,25 +99,30 @@ public CacheResult loadData(ProviderCache providerCache) {
     Collection<String> serverGroups =
         providerCache.filterIdentifiers(SERVER_GROUPS.ns, serverGroupsGlob);
     Collection<String> existing = providerCache.existingIdentifiers(SERVER_GROUPS.ns, serverGroups);
-    Set<String> clusterKeys =
-        existing.stream()
-            .map(keyParser)
-            .filter(Objects::nonNull)
-            .map(clusterKeyBuilder)
-            .collect(Collectors.toSet());
+    Map<String, Set<String>> clusterData = new HashMap<>();
+    for (String serverGroupKey : existing) {
+      Map<String, String> serverGroupData = keyParser.apply(serverGroupKey);
+      if (serverGroupData != null) {
+        String clusterKey = clusterKeyBuilder.apply(serverGroupData);
+        clusterData.computeIfAbsent(clusterKey, k -> new HashSet<>()).add(serverGroupKey);
+      }
+    }
 
-    Collection<CacheData> clusterData =
-        clusterKeys.stream()
+    Collection<CacheData> cacheData =
+        clusterData.entrySet().stream()
             .map(
-                s -> {
-                  Map<String, String> clusterAttributes = keyParser.apply(s);
-                  return new DefaultCacheData(
-                      s,
-                      Collections.singletonMap("name", clusterAttributes.get(clusterNameAttribute)),
-                      Collections.emptyMap());
+                entry -> {
+                  Map<String, String> clusterAttributes = keyParser.apply(entry.getKey());
+                  Map<String, Object> cacheAttributes = new HashMap<>();
+                  cacheAttributes.put("name", clusterAttributes.get(clusterNameAttribute));
+                  cacheAttributes.put(
+                      "application", clusterAttributes.get(clusterApplicationAttribute));
```

Comment:
this specifically was necessary because we use the getAllByApplication lookup to find the cluster objects